### DISCUSSION TITLE
面板自签ssl证书错误问题

### DISCUSSION THREAD

**Discussion by JAINKRE** [1Panel-dev/1Panel#6022] [Q&A]
### 1Panel 版本

v1.10.13-lts

### 问题描述

目前发现两个问题

## 火狐连接失败

使用火狐登录面板，面板开启ssl，报错：

![192f8fe7-c60f-4f01-806e-73bc149224b6](https://github.com/user-attachments/assets/36b2ab9e-8aee-4b02-93ec-8b9d5765df97)

火狐版本：128.0.3

使用谷歌浏览器可以登录：

![56b559bd-d5c7-4c24-ab44-c3736a4f87c3](https://github.com/user-attachments/assets/2428dd97-b056-42b6-b53b-8b9f70a0423f)

## 网站使用自签证书错误

新建默认静态网站：
![image](https://github.com/user-attachments/assets/e52d5897-1bcd-4aa1-a8b0-295b248f2c08)

开启SSL：
![f20f8eba-15e8-4e6e-82e6-c7a522645f27](https://github.com/user-attachments/assets/ec4eb724-8dea-4b8e-adae-50fad7126343)

访问失败：
![57b94d42-9377-4f41-a0a7-0c0dc6256be6](https://github.com/user-attachments/assets/19a0af41-52b3-4f96-94c1-475a16962b17)

谷歌浏览器：版本 127.0.6533.89（正式版本） （64 位）

### 重现步骤

使用默认自签证书出现上述问题，使用自己颁发证书也出现这个问题。

网站不仅静态网站启用https错误，反向代理也有同样问题

### 期待的正确结果

初步怀疑是SSL 协议版本问题

火狐浏览器错误提示：https://support.mozilla.org/zh-CN/kb/%E5%AE%89%E5%85%A8%E8%BF%9E%E6%8E%A5%E5%A4%B1%E8%B4%A5%E8%80%8C%20Firefox%20%E6%98%BE%E7%A4%BA%E9%94%99%E8%AF%AF?as=u&utm_source=inproduct#w_bu-zhi-chi-de-tls-ban-ben

**Comment by JAINKRE** [comment]
静态网站开启ssl后使用chrome和edge访问错误代码：ERR_SSL_PROTOCOL_ERROR

**Comment by zhengkunwang223** [comment]
> https://support.mozilla.org/zh-CN/kb/%E5%AE%89%E5%85%A8%E8%BF%9E%E6%8E%A5%E5%A4%B1%E8%B4%A5%E8%80%8C%20Firefox%20%E6%98%BE%E7%A4%BA%E9%94%99%E8%AF%AF?as=u&utm_source=inproduct#w_bu-zhi-chi-de-tls-ban-ben

![image](https://github.com/user-attachments/assets/c5f91e47-1c78-42d4-a111-7f27fe7a0c6d)
看看有没有这两行 有的话删掉

**Comment by JAINKRE** [comment]
> > https://support.mozilla.org/zh-CN/kb/%E5%AE%89%E5%85%A8%E8%BF%9E%E6%8E%A5%E5%A4%B1%E8%B4%A5%E8%80%8C%20Firefox%20%E6%98%BE%E7%A4%BA%E9%94%99%E8%AF%AF?as=u&utm_source=inproduct#w_bu-zhi-chi-de-tls-ban-ben
> 
> ![image](https://private-user-images.githubusercontent.com/31820853/354806668-c5f91e47-1c78-42d4-a111-7f27fe7a0c6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2ODkyODcsIm5iZiI6MTcyMjY4ODk4NywicGF0aCI6Ii8zMTgyMDg1My8zNTQ4MDY2NjgtYzVmOTFlNDctMWM3OC00MmQ0LWExMTEtN2YyN2ZlN2EwYzZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDEyNDMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMjk2OGY1YjgxNjBjNjkyOTA4NmU4YjAzN2I0NmQwODQ0NjQzZmZkM2IxYWNlNDlmZDE3ZjZmYjVmNzhjNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gVdfdI1GGE_RsfCeYBCAHcMIqHdKRNkRzoX80xSs0cA) 看看有没有这两行 有的话删掉

网站配置文件没有这两个参数，用的都是默认的：
![image](https://github.com/user-attachments/assets/2759ce70-ceb4-4bf5-a78a-f84d7215b8de)

**Comment by zhengkunwang223** [comment]
> > > https://support.mozilla.org/zh-CN/kb/%E5%AE%89%E5%85%A8%E8%BF%9E%E6%8E%A5%E5%A4%B1%E8%B4%A5%E8%80%8C%20Firefox%20%E6%98%BE%E7%A4%BA%E9%94%99%E8%AF%AF?as=u&utm_source=inproduct#w_bu-zhi-chi-de-tls-ban-ben
> > 
> > 
> > ![image](https://private-user-images.githubusercontent.com/31820853/354806668-c5f91e47-1c78-42d4-a111-7f27fe7a0c6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2ODkyODcsIm5iZiI6MTcyMjY4ODk4NywicGF0aCI6Ii8zMTgyMDg1My8zNTQ4MDY2NjgtYzVmOTFlNDctMWM3OC00MmQ0LWExMTEtN2YyN2ZlN2EwYzZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDEyNDMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMjk2OGY1YjgxNjBjNjkyOTA4NmU4YjAzN2I0NmQwODQ0NjQzZmZkM2IxYWNlNDlmZDE3ZjZmYjVmNzhjNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gVdfdI1GGE_RsfCeYBCAHcMIqHdKRNkRzoX80xSs0cA) 看看有没有这两行 有的话删掉
> 
> 网站配置文件没有这两个参数，用的都是默认的： ![image](https://private-user-images.githubusercontent.com/63288007/354817570-2759ce70-ceb4-4bf5-a78a-f84d7215b8de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2OTQyNjMsIm5iZiI6MTcyMjY5Mzk2MywicGF0aCI6Ii82MzI4ODAwNy8zNTQ4MTc1NzAtMjc1OWNlNzAtY2ViNC00YmY1LWE3OGEtZjg0ZDcyMTViOGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDE0MDYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1ODJmMzdiMDBmMTI3YzQzM2I3MGU3OGE2YzEzNzQzMWNiYWQ2NzdlNTdkMzhmNTczNjlmNTgyZWM2ZmJmMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.T3LhIXktrnYZFPQMcitP6R_RZYzf9YOwiYGP7kS97kw)

有 WAF 的话关闭  WAF

**Comment by JAINKRE** [comment]
> > > > https://support.mozilla.org/zh-CN/kb/%E5%AE%89%E5%85%A8%E8%BF%9E%E6%8E%A5%E5%A4%B1%E8%B4%A5%E8%80%8C%20Firefox%20%E6%98%BE%E7%A4%BA%E9%94%99%E8%AF%AF?as=u&utm_source=inproduct#w_bu-zhi-chi-de-tls-ban-ben
> > > 
> > > 
> > > ![image](https://private-user-images.githubusercontent.com/31820853/354806668-c5f91e47-1c78-42d4-a111-7f27fe7a0c6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2ODkyODcsIm5iZiI6MTcyMjY4ODk4NywicGF0aCI6Ii8zMTgyMDg1My8zNTQ4MDY2NjgtYzVmOTFlNDctMWM3OC00MmQ0LWExMTEtN2YyN2ZlN2EwYzZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDEyNDMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMjk2OGY1YjgxNjBjNjkyOTA4NmU4YjAzN2I0NmQwODQ0NjQzZmZkM2IxYWNlNDlmZDE3ZjZmYjVmNzhjNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gVdfdI1GGE_RsfCeYBCAHcMIqHdKRNkRzoX80xSs0cA) 看看有没有这两行 有的话删掉
> > 
> > 
> > 网站配置文件没有这两个参数，用的都是默认的： ![image](https://private-user-images.githubusercontent.com/63288007/354817570-2759ce70-ceb4-4bf5-a78a-f84d7215b8de.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI2OTQyNjMsIm5iZiI6MTcyMjY5Mzk2MywicGF0aCI6Ii82MzI4ODAwNy8zNTQ4MTc1NzAtMjc1OWNlNzAtY2ViNC00YmY1LWE3OGEtZjg0ZDcyMTViOGRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAzVDE0MDYwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE1ODJmMzdiMDBmMTI3YzQzM2I3MGU3OGE2YzEzNzQzMWNiYWQ2NzdlNTdkMzhmNTczNjlmNTgyZWM2ZmJmMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.T3LhIXktrnYZFPQMcitP6R_RZYzf9YOwiYGP7kS97kw)
> 
> 有 WAF 的话关闭 WAF

没有启用WAF，因为这个是内部用的服务器，WAF从没有开过
